### PR TITLE
uv__io_poll: try epoll_pwait if epoll_wait is missing

### PR DESCRIPTION
**Pull request by mwhudson**
It seems that epoll_wait is implemented in glibc in terms of epoll_pwait and
new architectures (like arm64) do not implement the epoll_wait syscall at all.
So if __NR_epoll_wait is undefined, we should just call uv__epoll_pwait.

An alternative to https://github.com/joyent/libuv/pull/1474.

### CODE REVIEW

File: src/unix/linux-core.c [hunk 1]
```diff
@@ -195,10 +196,18 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
...
-                          events,
-                          ARRAY_SIZE(events),
-                          timeout);
+    if (!no_epoll_wait) {
+      nfds = uv__epoll_wait(loop->backend_fd,
+                            events,
+                            ARRAY_SIZE(events),
+                            timeout);
+    } else {
+      nfds = uv__epoll_wait(loop->backend_fd,
```
Comment (saghul) [inline src/unix/linux-core.c]: I guess you mean pwait here?

File: src/unix/linux-core.c [hunk 2]
```diff
@@ -212,8 +221,13 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     }
 
     if (nfds == -1) {
-      if (errno != EINTR)
+      if (errno != EINTR) {
+	if (errno == ENOSYS && !no_epoll_wait) {
+          no_epoll_wait = 1;
```
Comment (saghul) [inline src/unix/linux-core.c]: Please move this test above, right after doing the epoll_wait.

**Comment by mwhudson** [comment]
Ooops.  Fixed, and amended as asked.

File: src/unix/linux-core.c [hunk 3]
```diff
@@ -195,10 +196,22 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
...
+      nfds = uv__epoll_wait(loop->backend_fd,
+                            events,
+                            ARRAY_SIZE(events),
+                            timeout);
+      if (nfds == -1 && errno == ENOSYS) {
+        no_epoll_wait = 1;
+        continue;
+      }
+    } else {
+      nfds = uv__epoll_wait(loop->backend_fd,
```
Comment (saghul) [inline src/unix/linux-core.c]: This needs to be pwait.

**Comment by mwhudson** [comment]
Aaargh git fail.  This time for sure?

**Comment by saghul** [comment]
Yeah, landed in 861de3d, thanks!

**Comment by mwhudson** [comment]
Yay, thanks!

Just curious really but I originally found this problem in node; will this fix appear there reasonably soon?

**Comment by saghul** [comment]
I'd say so, yes. I'm also adding it to the list of backports that will go on 0.10 as well.

**Pull request closed without merging**